File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        for num in nums:
            if num in count:
                count[num] += 1
            else:
                count[num] = 1
        arr = [[] for i in range(len(nums) + 1)]
        for key, value in count.items():
            arr[value].append(key)
        res = []
        for i in range(len(arr) - 1, 0, -1):
            for j in range(len(arr[i])):
                res.append(arr[i][j])
                k = k - 1
                if k == 0:
                    return res
        return res

File: Data Structures & Algorithms/count-number-of-islands/submission-1.cpp
class Solution {
public:
    int numIslands(vector<vector<char>>& grid) {
        int count = 0;
        int n = grid.size();
        int m = grid[0].size();
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (grid[i][j] == '2') {
                    continue;
                }
                if (grid[i][j] == '1') {
                    if (dfs(grid, i, j, n, m)) {
                        count += 1;
                    }
                }
            }
        }
        return count;
    }

    bool dfs(vector<vector<char>>& grid, int i, int j , int n, int m) {
        if (i < 0 || i >= n || j < 0 || j >= m) {
            return true;
        }
        if (grid[i][j] == '0') {
            return true;
        }
        else if (grid[i][j] == '1') {
            grid[i][j] = '2';
            return dfs(grid, i + 1, j, n, m) && 
                    dfs(grid, i - 1, j, n, m) &&
                    dfs(grid, i, j - 1, n, m) && 
                    dfs(grid, i, j + 1, n, m);
        }
        else {
            return true;
        }
    }
};
